docs/migrations-guidelines.md

```diff
@@ -0,0 +1,89 @@
+# Migration and Testing Guidelines
+
+## Overview
+
+This document outlines best practices and guidelines for writing migration scripts and their corresponding test cases.
+
+The focus is on ensuring data integrity, handling errors gracefully, and maintaining consistency across mobile app versions.
+
+We always look for improvement, if you see anything that could be improved, please open a PR against this guidelines.
+
+You can also check an example of a migration on MetaMask mobile app [here](https://github.com/MetaMask/metamask-mobile/blob/1855bd674e33bb0ece06fb6d8f09a4e5df46a108/app/store/migrations/044.ts#L1)
+
+## Migration Guidelines
+
+1. **Pre-Validation Checks**:
+
+- Ensure the state to be migrated is valid before running the migration logic,
+- Verify the version and integrity of the state with utility functions like `ensureValidState`.
+
+2. **Error Handling**:
+
+The following prevents manipulation of potentially corrupted data, enforcing corrective action before proceeding.
+
+- Log erros with `captureException` from `@sentry/react-native`, which is crucial for diagnosing issues post-migration,
+- Ensure that error messages are descriptive: include the migration number and a clear description of the issue,
+- If an exception is detected, indicating potential data corruption, halt the migration process and return the intial state,
+
+3. **State Integrity Checks**:
+
+- Checks the state's structure and types carefully,
+- Validate the state and its nested properties using functions like `isObject` and `hasProperty`,
+- Prevent data corruption by halting the migration on any inconsistencies and logging errors.
+
+5. **Avoid Type Casting**:
+
+- Do not use type casting (for example, `as` keyword in TypeScript) to manipulate the state or its properties,
+- Type casting can mask underlying data structure issues and make the code less resilient in case of change in dependencies or data structures,
+- Instead, use type guards and runtime checks to ensure data integrity and compatibility.
+
+6. **Return State**:
+
+- Always return the state at the end of the migration function, whether it was modified or not,
+- Returning the state ensures that the migration process completes and the state is passed to the next migrations.
+
+## Testing Guidelines
+
+1. **Mocking**:
+
+- Mock dependencies with `jest.mock` (for example `@sentry/react-native`),
```

Comment:
Perhaps we could focus these guidelines to be more specific for migrations. Much of these steps seem applicable to all unit tests